optimum/habana/transformers/generation/stopping_criteria.py

```diff
@@ -52,18 +52,6 @@ def gaudi_MaxLengthCriteria_call(
         return create_return_const_tensor(input_ids, is_done)
 
 
-def gaudi_MaxNewTokensCriteria_call(
-    self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs
-) -> Union[torch.BoolTensor, bool]:
-    token_idx = kwargs.get("token_idx", None)
-    if token_idx is not None:
-        assert not kwargs["needs_tensor_output"]
-        return token_idx >= self.max_length
-    else:
-        is_done = input_ids.shape[-1] >= self.max_length
-        return create_return_const_tensor(input_ids, is_done)
```

Comment:
@libinta MaxNewTokensCriteria no longer exists in transformers.

removed in this PR: https://github.com/huggingface/transformers/pull/32659/files#diff-6e63ae0764aa864afd5bae6d512677b99b5240cb98cb210190482bdbb6a85906


It was removed as it had plans for being deprecated:

"The class `MaxNewTokensCriteria` is deprecated and will be removed in v4.43. "
            f"Please use `MaxLengthCriteria(max_length={start_length + max_new_tokens})` "